core/src/block_production_manager.rs

```diff
@@ -0,0 +1,156 @@
+use {
+    crate::{
+        banking_stage::{committer::Committer, vote_storage::VoteStorage, BankingStage},
+        validator::{BlockProductionMethod, TransactionStructure},
+    },
+    agave_banking_stage_ingress_types::BankingPacketReceiver,
+    solana_ledger::blockstore_processor::TransactionStatusSender,
+    solana_poh::{poh_recorder::PohRecorder, transaction_recorder::TransactionRecorder},
+    solana_runtime::{
+        bank_forks::BankForks, prioritization_fee_cache::PrioritizationFeeCache,
+        vote_sender_types::ReplayVoteSender,
+    },
+    std::{
+        sync::{
+            atomic::{AtomicBool, Ordering},
+            Arc, RwLock,
+        },
+        thread::{self, JoinHandle},
+    },
+};
+
+/// Handle to manage block production.
+pub struct BlockProductionManager {
```

Comment:
this actually gives us some flexibility to only keep threads alive if we're a staked node 🤔

not gonna deal with that in this PR. but we could make it so RPCs don't run block-production threads at all.